embedding/csharp/Tizen.Flutter.Embedding/NUIFlutterView.cs

```diff
@@ -0,0 +1,211 @@
+        /// </summary>
+        private Size2D GetDefaultSize()
+        {
+            if (Size2D.Width == 0 || Size2D.Height == 0)
+            {
+                Container parent = GetParent();
+                if (parent == null)
+                {
+                    Layer layer = Window.Instance.GetDefaultLayer();
+                    return new Size2D(layer.Viewport.Width, layer.Viewport.Height);
+                }
+
+                if (parent is View)
+                {
+                    View view = parent as View;
+                    return view.Size2D;
+                }
+                else if (parent is Layer)
+                {
+                    Layer layer = parent as Layer;
+                    return new Size2D(layer.Viewport.Width, layer.Viewport.Height);
+                }
+            }
+            return Size2D;
+        }
+
+        /// <summary>
+        /// Registers view event handlers.
+        /// </summary>
+        private void RegisterEventHandlers()
+        {
+            Focusable = true;
+            KeyEvent += (object s, KeyEventArgs e) =>
+            {
+                if (!IsRunning)
+                {
+                    return true;
+                }
+
+                FlutterDesktopViewOnKeyEvent(
+                    View, e.Key.KeyPressedName, e.Key.KeyPressed, (uint)e.Key.KeyModifier, (uint)e.Key.KeyCode,
+                    e.Key.State == Key.StateType.Down);
+                return true;
+            };
+
+            TouchEvent += (object s, TouchEventArgs e) =>
+            {
+                if (!IsRunning)
+                {
+                    return true;
+                }
+
+                if (_lastTouchEventTime == e.Touch.GetTime())
+                {
+                    return false;
+                }
+                FocusManager.Instance.SetCurrentFocusView(this);
+
+                FlutterDesktopPointerEventType type;
+                switch (e.Touch.GetState(0))
+                {
+                    case PointStateType.Down:
+                    default:
+                        type = FlutterDesktopPointerEventType.kPointerDown;
+                        break;
+                    case PointStateType.Up:
+                        type = FlutterDesktopPointerEventType.kPointerUp;
+                        break;
+                    case PointStateType.Motion:
+                        type = FlutterDesktopPointerEventType.kPointerMove;
+                        break;
+                }
+                FlutterDesktopViewOnPointerEvent(
+                    View, type, e.Touch.GetLocalPosition(0).X, e.Touch.GetLocalPosition(0).Y, e.Touch.GetTime(),
+                    e.Touch.GetDeviceId(0));
+
+                _lastTouchEventTime = e.Touch.GetTime();
+                return true;
+            };
+
+            Relayout += (object s, EventArgs e) =>
+            {
+                if (IsRunning && (_size.Width != Size2D.Width || _size.Height != Size2D.Height))
```

Comment:
> As far as I know that when I call SetImage(), the ImageView's Size2D is set to the target's Size.
However, this size is set when Relayout() is called.
I modified it to set Size after SetImage.

How about moving this check to the native side (`FlutterTizenView::Resize`) so that you don't have to manage the `_size` value?

Reply:
I moved the size check code to engine.
https://github.com/flutter-tizen/engine/pull/328/commits/69d461813b19e749581d48b71f7bdc4dfd676f7b

Reply:
> I moved the size check code to engine. [flutter-tizen/engine@69d4618](https://github.com/flutter-tizen/engine/commit/69d461813b19e749581d48b71f7bdc4dfd676f7b)

I reverted the engine code and changed the embedding to check the size.
(https://github.com/flutter-tizen/engine/pull/328/commits/7bfd4ea1655e05345d3c60b05d358c71dc2ae01c)